tempesta_fw/http_limits.c

```diff
@@ -332,16 +319,25 @@ __frang_init_acc(void *data)
 static int
 frang_conn_new(struct sock *sk)
 {
-	int r;
+	int r = TFW_BLOCK;
 	FrangAcc *ra;
 	TfwClient *cli;
-	FrangCfg *conf = tfw_vhost_global_frang_cfg();
 	TfwAddr addr;
+	TfwVhost *dflt_vh = tfw_vhost_get_default();
+
+	/*
+	 * On reboot under heavy load global vhost may be not ready yet.
+	 * Don't pass such connections, force client to reconnect later, when
+	 * we will be ready.
+	 */
+	if (unlikely(!dflt_vh))
+		return TFW_BLOCK;
```

Comment:
> Also consider that we can have vhost from TLS SNI 
I was thinking about it a lot, but it turns out that the feature is leading to rather unpredictable behaviour. Too many factors are taking into account when it's decided block client or not.

I.e. a client opens a few TLS connection to vhost `a.com` (allowed), then a thousands to `b.com` (forbidden). Should we block it? Should we block all connection or only connections to `b.com`? What if `ip_block` option set? Probably we should block only connections to vhost `b.com` and remain others to be active. I don't have a good answer.

Same situation, but client opens simple TCP connection (not TLS). We can't distinguish connections to `a.com` from `b.com`, and all will be closed on security events. 

Same situation as first one, but SAN certificate is configured as default and only certificate. (One certificate that covers all the protected vhosts, pretty common for CDNs as I got it). In such case the default vhost is chosen for both `a.com`, `b.com` at TLS level (questionable, but it's the #1274 question). And again no possibility to distinguish one connections from another.

Looks pretty simple on papers, but it will be hard to debug and investigate. I believe, that it's possible to make all limits really per-vhost in TLS word, but not now. Too complicated and unpredictable. But it will be very reliable and understandable for machine-learning QoS and filters.

> Please review nftables and iptables abilities in TCP connection limitation: connection rate with burst control and parallel connections from the same client. 

Both iptables and nftables support it:  https://wiki.nftables.org/wiki-nftables/index.php/Quick_reference-nftables_in_10_minutes#Limit 

At this stage we can remove the hooks. But in the future (machine-learning filtering I've mentioned above), we will need them again to provide different blocking facilities for different vhost. With widespread http/2 it will be more handy. Probably. couldn't decide and left as is.

Reply:
OK, I think we can leave the connection limits and ignore SNI for now.